crates/astria-sequencer/src/app/tests_breaking_changes.rs

```diff
@@ -307,3 +351,175 @@ async fn app_execute_transaction_with_every_action_snapshot() {
+                }
+                .into(),
+                Action::ValidatorUpdate(ValidatorUpdate {
+                    power: 100,
+                    verification_key: crate::test_utils::verification_key(1),
+                }),
+                BridgeLockAction {
+                    to: bridge_address,
+                    amount: 100,
+                    asset: nria().into(),
+                    fee_asset: nria().into(),
+                    destination_chain_address: "nootwashere".to_string(),
+                }
+                .into(),
+                BridgeUnlockAction {
+                    to: bob_address,
+                    amount: 10,
+                    fee_asset: nria().into(),
+                    memo: String::new(),
+                    bridge_address: astria_address(&bridge.address_bytes()),
+                    rollup_block_number: 1,
+                    rollup_withdrawal_event_id: "a-rollup-defined-hash".to_string(),
+                }
+                .into(),
+                Ics20Withdrawal {
+                    amount: 1,
+                    denom: nria().into(),
+                    bridge_address: None,
+                    destination_chain_address: "test".to_string(),
+                    return_address: bob_address,
+                    timeout_height: Height::new(1, 1).unwrap(),
+                    timeout_time: 1,
+                    source_channel: "channel-0".to_string().parse().unwrap(),
+                    fee_asset: nria().into(),
+                    memo: String::new(),
+                    use_compat_address: false,
+                }
+                .into(),
+            ])
+            .chain_id("test")
+            .try_build()
+            .is_ok(),
+        "should be able to construct general bundle"
+    );
+
+    assert!(
+        UnsignedTransaction::builder()
+            .actions(vec![
+                IbcRelayerChangeAction::Addition(bob_address).into(),
+                FeeAssetChangeAction::Removal("test-0".parse().unwrap()).into(),
+                FeeChangeAction {
+                    fee_change: FeeChange::TransferBaseFee,
+                    new_value: 10,
+                }
+                .into(),
+            ])
+            .nonce(1)
+            .chain_id("test")
+            .try_build()
+            .is_ok(),
+        "should be able to construct sudo bundle"
+    );
+
+    assert_eq!(
+        UnsignedTransaction::builder()
+            .actions(vec![
+                SudoAddressChangeAction {
+                    new_address: bob_address,
+                }
+                .into(),
+                SudoAddressChangeAction {
+                    new_address: bob_address,
+                }
+                .into(),
+            ])
+            .nonce(2)
+            .chain_id("test")
+            .try_build()
+            .unwrap_err()
+            .to_string(),
+        "attempted to create bundle with non bundleable `ActionGroup` type: unbundleable sudo"
```

Comment:
As before, I think these tests should match on the `ErrorKind` variant rather than on the `Display` output of the error.

Reply:
I made the whole `ActionGroup` enum public in the process to achieve this.